packs/signoz/README.md

```diff
@@ -0,0 +1,166 @@
+# SigNoz on HashiCorp Nomad
+
+Deploy the complete [SigNoz](https://signoz.io) observability stack on HashiCorp Nomad. This pack includes ClickHouse, ZooKeeper, the SigNoz application, and the OpenTelemetry Collector, all running as Docker containers via Nomad's [Docker driver](https://www.nomadproject.io/docs/drivers/docker).
+
+## Prerequisites
+
+Ensure your environment has:
+
+- A running Nomad cluster
+- Consul enabled for service discovery
+- Nomad variable store configured
+- Host volumes set up for persistent storage
+
+## Setup Guide
+
+### Step 1: Configure ClickHouse Password
+
+Store your ClickHouse password securely using Nomad variables:
+
+```hcl
+# spec.nv.hcl
+# Use path "nomad/jobs" to make it accessible to all jobs in the pack
+path = "nomad/jobs/<your_pack_release_name>"
+
+items {
+  clickhouse_password = "your_clickhouse_password"  # Change this
+}
+```
+
+Load the variables:
+
+```bash
+nomad var put @spec.nv.hcl
+```
+
+### Step 2: Create Access Policy
+
+Define a policy that grants tasks read access to your variables:
+
+```hcl
+# signoz-shared-vars.policy.hcl
+namespace "<your-shared-namespace>" {
+  variables {
+    path "nomad/jobs/<your_release_name>" {
```

Comment:
This would make this variable available to any _other_ job named `nomad/jobs/<your release name>` automatically without applying an ACL policy. If it's not going to be a workload-associated path, it should probably be an independent path under somewhere other than `nomad/jobs`.

Reply:
Okay, since the path isn’t workload-specific, I’ll move it to an independent path. Also, I don’t think Nomad lets us define policies based on selectors or annotations, or any other way that might fit our use case, right? If that’s the case, we can’t avoid making it available to all jobs under the shared namespace.